api/v1/postgres_types.go

```diff
@@ -327,11 +325,6 @@ func (p *Postgres) ToUserPasswordsSecret(src *corev1.SecretList, scheme *runtime
 		secret.Data[string(v.Data["username"])] = v.Data["password"]
 	}
 
-	// Set the owner of the secret
```

Comment:
What was this controller reference for? Automatic deletion?
And why can we remove it now?

Reply:
`func SetControllerReference` is for [garbage collection](https://github.com/LimKianAn/xcluster#controllerreference). However, it can't be tested in [envtest](https://cluster-api.sigs.k8s.io/developer/testing.html).
> One common example is garbage collection; because there are no controllers monitoring built-in resources, objects do not get deleted, even if an OwnerReference is set up; as a consequence, usually test implements code for cleaning up created objects.

So an explicit deletion was implemented.

Reply:
Ah, ok. The controller is running in another cluster anyway.